src/docs/tree_widget.md

```diff
@@ -0,0 +1,723 @@
+---
+title: Tree Widget
+---
+
+# The Tree Widget
+
+Interacting with hierarchical content is an integral part of many applications. Tree UIs are a common aspect of many user interfaces. In Theia, the `@theia/core` module provides `TreeWidget`, a common implementation of tree UIs in Theia, which can be customized in many ways to let the user work with hierarchical content. This article provides an overview of `TreeWidget`, its related classes and features, and how to implement and customize them.
+
+Note: The code snippets shown below are part of the full example available via the [Theia Extension Generator](https://github.com/eclipse-theia/generator-theia-extension/blob/master/README.md). To get started and play with the example code, generate the `TreeWidget View` example using the generator.
+
+## Basic Building Blocks
+
+To create a basic `TreeWidget`, we need to implement:
+
+* the actual widget - a subclass of `TreeWidget`
+* a model - a subclass of `TreeModelImpl`
+* one or more tree node interfaces (extending `TreeNode` or its sub-interfaces)
+* a label provider contribution - a subclass of `LabelProviderContribution`
+* the view contribution - a subclass of `AbstractViewContribution`, as is usual for all widgets/views
+* the widget factory, which is realized using an inversify child container
+
+All of these are explained below in more detail. For demonstration purposes, we will create a `TreeviewExampleWidget` with some demo contents.
+
+### The `TreeWidget` widget
+
+The `TreeWidget` is a specialized `ReactWidget` that already implements all of the logic required to render trees and let the user interact with them. So in its simplest form, we need to just implement the constructor to initialize the id, title, and caption
```

Comment:
I am not sure if we "need" to do it but we definitely want to do it. I'd be a bit more careful in what we describe as necessary in this article vs a common approach/recommendation.